### ISSUE TITLE
[Bug]: How to change entities in the predict function

### ISSUE THREAD

**Issue by JINO-ROHIT** [zalandoresearch/flair#3675]
### Describe the bug

I need to detect only PERSON AND EMAIL instead of all the 2o entities.how to change this?

Loading model from flair/ner-english-large
2025-06-16 22:55:27,369 SequenceTagger predicts: Dictionary with 20 tags: <unk>, O, S-ORG, S-MISC, B-PER, E-PER, S-LOC, B-ORG, E-ORG, I-PER, S-PER, B-MISC, I-MISC, E-MISC, I-ORG, B-LOC, E-LOC, I-LOC, <START>, <STOP>

### To Reproduce

```python
from flair.data import Sentence
from flair.nn import Classifier

# make a sentence
sentence = Sentence('George Washington went to Washington.')

# load the NER tagger
tagger = Classifier.load('ner-large')

# run NER over sentence
tagger.predict(sentence)

# print the sentence with all annotations
print(sentence)
```

### Expected behavior

NA

### Logs and Stack traces

```stacktrace

```

### Environment

NA

**Comment by JINO-ROHIT** [comment]
@helpmefindaname

**Comment by hsleonis** [comment]
This isn't a bug. Pre-trained NER models like `ner-english-large` are designed to detect all entity types they were trained on (in this case, 20 types). You can't modify what entities the model predicts at runtime.
Solution: Filter the results after prediction to keep only **PERSON** entities:

````python
from flair.data import Sentence
from flair.nn import Classifier

sentence = Sentence('George Washington went to Washington.')
tagger = Classifier.load('ner-large')
tagger.predict(sentence)

# Filter for only PERSON entities
person_entities = [entity for entity in sentence.get_spans('ner') 
                   if 'PER' in entity.tag]
print(person_entities)
````

For EMAIL detection, you'll need a different approach since standard NER models don't typically detect email addresses - consider using regex or training a custom model.

**Comment by jbbqqf** [comment]
Hi! Drive-by triage of older issues — I believe this can be closed as a usage question that's already been answered.

The premise ("how do I limit `ner-english-large` to only PERSON / EMAIL?") doesn't have a model-side answer: a pre-trained NER tagger predicts exactly the entity types its training data carries. `@hsleonis` already gave the right pattern in this thread — filter the spans after `predict()`.

Evidence:
- `flair.data.Sentence.get_spans()` is the documented accessor for span-level predictions: https://github.com/flairNLP/flair/blob/d4ea3777998ba67bfbe6b6b8359e024dcf673c3e/flair/data.py#L1442
- The official tutorial "How predictions work" shows the exact iterate-and-inspect pattern (`for label in sentence.get_labels(): ... label.value ...`): https://github.com/flairNLP/flair/blob/d4ea3777998ba67bfbe6b6b8359e024dcf673c3e/docs/tutorial/tutorial-basics/how-predictions-work.md#L37-L51
- The `SequenceTagger` docstring example iterates spans the same way: https://github.com/flairNLP/flair/blob/d4ea3777998ba67bfbe6b6b8359e024dcf673c3e/flair/models/sequence_tagger_model.py#L832
- For EMAIL: no Flair pre-trained model emits an `EMAIL` tag (it's not in OntoNotes/CoNLL label sets), so post-processing with a regex or a custom-trained tagger is the path — not a model-config knob.

If I'm missing a flag on `predict()` you actually meant, point me at it and I'll keep digging. Otherwise would you mind closing this out?

---
*Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the references above were verified manually.*
